### PR TITLE
JNI api for cudf::chunked_pack

### PR DESCRIPTION
**Pull request by abellina**
This PR is the standalone JNI side of https://github.com/rapidsai/cudf/pull/13260. Therefore it doesn't build in as is, but I am putting this up as draft for the Java reviewers to start taking a look.

This implements a `ChunkedPack` java class mirroring the interface of `cudf::chunked_pack`. This is an iterator-like class that can be used to invoke `cudf::pack` (aka `cudf::contigous_split` without splits) over several iterations against a bounce buffer.

In order to create a `ChunkedPack`, the user calls `makeChunkedPack` from a `Table` instance. During this call the user can also pass an `RmmDeviceMemoryResource` to be used internally by `cudf::chunked_pack` exclusively for scratch/temporary data.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ChunkedPack.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ */
+
+package ai.rapids.cudf;
+
+/**
+ * JNI interface to cudf::chunked_pack.
+ * 
+ * ChunkedPack is an iterator-like interface with the familiar `hasNext` and `next`
+ * interface. `next` should be used in a loop until `hasNext` returns false.
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ChunkedPack.java:24]: nit: can we say more like 
```
`ChunkedPack` has an Iterator-like API with the familiar `hasNext` and `next` methods.
```

File: java/src/main/native/src/PackedColumnMetadataJni.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  // Convert local reference to global so it cannot be garbage collected.
+  Packed_columns_meta_jclass = static_cast<jclass>(env->NewGlobalRef(cls));
+  if (Packed_columns_meta_jclass == nullptr) {
+    return false;
+  }
+  return true;
+}
+
+jobject packed_column_metadata_from(JNIEnv *env, std::unique_ptr<std::vector<uint8_t>> meta) {
```
Comment (revans2) [inline java/src/main/native/src/PackedColumnMetadataJni.cpp:52]: Why do we create a jobect and do all of the reflection/etc if we could just return a jlong to java and let it call the constructor in java?

File: java/src/main/java/ai/rapids/cudf/ChunkedPack.java (hunk 1 above)
Comment (abellina) [reply java/src/main/java/ai/rapids/cudf/ChunkedPack.java]: done

File: java/src/main/native/src/PackedColumnMetadataJni.cpp (hunk 2 above)
Comment (abellina) [reply java/src/main/native/src/PackedColumnMetadataJni.cpp]: yeap that is much easier, fixed.

**Review by jbrennan333 — COMMENTED** [review]
This looks good.

**Review by revans2 — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]

File: java/src/main/native/src/CudfJni.cpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019-2022, NVIDIA CORPORATION.
+ * Copyright (c) 2019-2023, NVIDIA CORPORATION.
```
Comment (revans2) [inline java/src/main/native/src/CudfJni.cpp:2]: Is this needed? We are only changing the copyright date???

File: java/src/main/native/src/cudf_jni_apis.hpp [hunk 4]
```diff
@@ -51,7 +51,6 @@ convert_table_for_return(JNIEnv *env, std::unique_ptr<cudf::table> &&table_resul
 //
 
 bool cache_contiguous_table_jni(JNIEnv *env);
-
```
Comment (revans2) [inline java/src/main/native/src/cudf_jni_apis.hpp:54]: why is this change needed?

**Review by revans2 — APPROVED** [review]
Just two curious changes that appear to not be needed.

File: java/src/main/native/src/CudfJni.cpp (hunk 3 above)
Comment (abellina) [reply java/src/main/native/src/CudfJni.cpp]: Sorry about that, it is fixed now.

File: java/src/main/native/src/cudf_jni_apis.hpp (hunk 4 above)
Comment (abellina) [reply java/src/main/native/src/cudf_jni_apis.hpp]: Fixed
